### PR TITLE
Delete coord used for grouping by `bin`

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This event coord is unnecessary since the bin-coord contains the full
information (unlike when *edges* are used, where the event coord is
necessary). If we preserve it there are a number of "issues":

- Wastes memory.
- Has to be processed in follow-up operations (including calling `bin`
another time), which can be costly.
- Inconsistent with `groupby`, where the grouping labels are dropped in
the grouping operation.

Note that this showed up in practice in `scippneutron.load_nexus`, which used `sc.bin` to group by `detector_id`, but preserved the IDs for every event.

### CODE REVIEW

File: dataset/bin.cpp
Comment (jl-wynen) [inline dataset/bin.cpp:200]: Does `concatenate` restore the coord correctly?

Comment (SimonHeybrock) [reply dataset/bin.cpp:200]: No, and in fact I did not understand the point of the original comment any more (and removed it): When you concatenate data arrays (or bins of data arrays) you still have the bin coords, so there is no reversibility issue.

The comment *might* come from the idea that someone could concatenate binned *variables* (rather than data arrays containing binned data), but this seems rare, and the lack of coords is just what you get from working with variables.

Comment (SimonHeybrock) [reply dataset/bin.cpp:200]: To be a bit more concrete: The comment may have been written with a reduction operation in mind:
```python
da.bins.concatenate('pixel')  # after this PR we don't know which event is from which pixel
```
But consider the "equivalent" for dense data:
```python
da.sum('pixel')  # don't know which contribution is from which pixel
```

*Iff* there is a user requirement to preserve this a flag could be added.

**Review by jl-wynen — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
Actually it would be more efficient to erase this coord *before* binning, don't merge yet.

**Comment by jl-wynen** [comment]
> Iff there is a user requirement to preserve this a flag could be added.

I recently ran into a situation where I had to reconstruct event data from bins including a group coordinate that was dropped manually. I had to build the events by hand. It would have been nice to have a function in scipp that can do it.
For context, it was data loaded from nexus which was binned in tof and grouped in pixel-id but the latter existed only for bins as proposed in this PR. I needed the events with associated tof and pixel-id for each event.

**Comment by SimonHeybrock** [comment]
> > Iff there is a user requirement to preserve this a flag could be added.
> 
> I recently ran into a situation where I had to reconstruct event data from bins including a group coordinate that was dropped manually. I had to build the events by hand. It would have been nice to have a function in scipp that can do it.
> For context, it was data loaded from nexus which was binned in tof and grouped in pixel-id but the latter existed only for bins as proposed in this PR. I needed the events with associated tof and pixel-id for each event.

We ran into this already a while back and added a mechanism to do so, based on a simple broadcast using `[...]`:

```python
da.events.coords['time'] = sc.empty(sizes=da.events.sizes, dtype='int64', unit='s')
da.bins.coords['time'][...] = da.coords['time']
```
(see https://scipp.github.io/user-guide/binned-data/binned-data.html#Manipulating-bin-based-and-event-based-metadata)

**Comment by jl-wynen** [comment]
Cool, did not see that. Thanks for the pointer!

**Comment by SimonHeybrock** [comment]
@jl-wynen Can you have another look. Cheers!

File: dataset/bin.cpp
Comment (jl-wynen) [inline dataset/bin.cpp:588]: Good idea! Is this also needed here? https://github.com/scipp/scipp/blob/8a68869cf45afe912f750dd7a4c20dfe6b0c0e6d/dataset/bin.cpp#L480-L482

Comment (SimonHeybrock) [reply dataset/bin.cpp:588]: I think not, see if the comment I just added explains.

Comment (jl-wynen) [reply dataset/bin.cpp:591]: I think that makes sense

**Pull request merged**
